src/AbstractDoctrineAnnotationFixer.php

```diff
@@ -207,13 +207,19 @@ protected function createConfigurationDefinition(): FixerConfigurationResolverIn
 
     private function nextElementAcceptsDoctrineAnnotations(Tokens $tokens, int $index): bool
     {
+        $classModifiers = [T_ABSTRACT, T_FINAL];
+
+        if (\defined('T_READONLY')) { // @TODO: drop condition when PHP 8.2+ is required
```

Comment:
`T_READONLY` was introduced when `readonly` for properties was introduced, but here this `@TODO` refers to 8.2 where `readonly` classes were added. I am not sure if it should be allowed to add this token to `$classModifiers` on 8.1 🤔

Reply:
> here this `@TODO` refers to 8.2 where `readonly` classes were added

This comment refers to `defined('T_READONLY')` check, which always evaluates to `true` on PHP >=8.1.

> I am not sure if it should be allowed to add this token to $classModifiers on 8.1 🤔

It doesn't matter, the logic is OK there.

Reply:
I know that condition itself is based on `T_READONLY` which is available in 8.1, but `@TODO` does not refer strictly to condition, rather to logic behind that condition (which operates on class modifiers, and `readonly` in this context is available on 8.2+). Maybe condition is wrong then 🤷‍♂️

Reply:
>Maybe condition is wrong then

can you check maybe?

Reply:
Maybe `version_compare()` or `PHP_VERSION_ID` should be used instead, so new modifier would be added only for 8.2+? 🤔

Reply:
should it not also apply to `readonly` class properties?

Reply:
For properties it should check PHP8.1+, as it was added earlier ([rfc](https://wiki.php.net/rfc/readonly_properties_v2)).

Reply:
- the PR checks the existence of the token
- review raises concern if it should be PHP version check and not token

So why do we need to check the PHP version and not the existence of the token? I know when it was added, I just don't understand the problem. Maybe someone can make unit test to show the problem of the code?

Reply:
I don't see any problem with the code, just think that the check can be deleted when we require PHP 8.1 and suggested updating the comment :)

Reply:
I don't see a problem either 😝 Technically it probably would be possible to remove condition on 8.1 because it's the same token for class and properties, right? But logically, there's no `readonly` modifier for class 🤷‍♂️ I don't have strong opinion here.

Reply:
ok, thanks all, I leave as is for now as it doesn't matter if dropped on 8.1 or 8.2 as the tests show